### PR TITLE
VACMS-10053: Move PHPUnit unit tests to separate GitHub Action.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Relates to #10053.

This PR does the following:
- ensures every PHPUnit class has certain group annotations, e.g. either `functional` or `unit` as appropriate, and `all`.
- prevents the Tugboat `va/tests/phpunit` action from executing `unit` tests
- adds a new GitHub Action to execute `unit` tests

## Notes

This has implications for documentation, but I believe it is actually very difficult to update `testing.md` to reflect this change without completely restructuring and rewriting that file.  I've opened a separate issue (#10558) to encompass that work.

## QA steps

If both existing (in-Tugboat) and new (GHA) tests pass, this should be good to go.

## Followup

- [ ] After merging, this needs to be added to branch protection rules.

### CODE REVIEW

File: tests/phpunit/API/PostApiQueueTest.php [hunk 1]
```diff
@@ -152,9 +153,9 @@ protected function setExpiration($queue, $expire) {
    */
   protected function processItem($data) {
     $apikey = Settings::get('post_api_apikey');
-    $endpoint_path = isset($data['endpoint_path']) ? $data['endpoint_path'] : NULL;
+    $endpoint_path = $data['endpoint_path'] ?? NULL;
     $endpoint = Settings::get('post_api_endpoint_host') . $endpoint_path;
-    $payload = isset($data['payload']) ? $data['payload'] : [];
+    $payload = $data['payload'] ?? [];
```
Comment (ndouglas) [inline tests/phpunit/API/PostApiQueueTest.php:158]: Not strictly related to PR topic, but PHPCS yelled at me until I converted this to null-coalescing operators.

**Comment by ndouglas** [comment]
**Before**
(from [another PR](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/10491)'s Tugboat PHPUnit run)

Screen Shot 2022-08-31 at 12 41 56 PM

**After**:
(Tugboat)
Screen Shot 2022-08-31 at 12 43 31 PM

(GitHub Action)
Screen Shot 2022-08-31 at 12 44 13 PM

**Comment by ndouglas** [comment]
```
  1) CMS Users may effectively create & edit content
       Log in and confirm that System-wide alerts can be created and edited:
     TypeError: Drupal.toolbar.ToolbarModel is not a constructor
      at https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/modules/toolbar/js/toolbar.js?v=9.4.5:28:21
      at Array.forEach (<anonymous>)
      at Object.attach (https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/modules/toolbar/js/toolbar.js?v=9.4.5:27:59)
      at https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.js?v=9.4.5:27:24
      at Array.forEach (<anonymous>)
      at Object.Drupal.attachBehaviors (https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.js?v=9.4.5:24:34)
      at https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.init.js?v=9.4.5:29:12
      at HTMLDocument.listener (https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.init.js?v=9.4.5:17:7)

  2) CMS Users may effectively create & edit content
       "after each" hook for "Log in and confirm that System-wide alerts can be created and edited":
     TypeError: Drupal.toolbar.ToolbarModel is not a constructor

Because this error occurred during a `after each` hook we are skipping the remaining tests in the current suite: `CMS Users may effectively c...`
      at https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/modules/toolbar/js/toolbar.js?v=9.4.5:28:21
      at Array.forEach (<anonymous>)
      at Object.attach (https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/modules/toolbar/js/toolbar.js?v=9.4.5:27:59)
      at https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.js?v=9.4.5:27:24
      at Array.forEach (<anonymous>)
      at Object.Drupal.attachBehaviors (https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.js?v=9.4.5:24:34)
      at https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.init.js?v=9.4.5:29:12
      at HTMLDocument.listener (https://pr10541-phrwrw5zlfz9jcdyk2pxglej2dhma3rz.ci.cms.va.gov/core/misc/drupal.init.js?v=9.4.5:17:7)
```

what?

**Review by cweagans — APPROVED** [review]
😍

**Pull request merged**